### PR TITLE
Add product edit endpoint

### PR DESCRIPTION
**Pull request by jeroen-hso**

### CODE REVIEW

**Comment by ahinkle** [comment]
Hey @jeroen-hso!

Thanks for submitting this PR. Would you mind adding a test for this? otherwise, I'll add it once I have the time. 

Thanks!!

File: tests/Api/ProductTest.php [hunk 1]
```diff
@@ -30,4 +30,15 @@ public function test_can_call_magento_api_products_show()
 
         $this->assertTrue($api->ok());
     }
+
+    public function test_can_edit_product()
+    {
+        Http::fake([
+            '*rest/default/V1/products/foo' => Http::response([], 200),
```
Comment (ahinkle) [inline tests/Api/ProductTest.php:37]:
```suggestion
            '*rest/default/V1/products/foo*' => Http::response([], 200),
```
For tests to pass--
I think GH Actions has some trailing slash or something going on. :)

**Review by ahinkle — CHANGES_REQUESTED** [review]

File: tests/Api/ProductTest.php (hunk 1 above)
Comment (ahinkle) [inline tests/Api/ProductTest.php:37]:
```suggestion
            '*rest/all/V1/products/foo' => Http::response([], 200),
```

for tests to pass :)

**Review by ahinkle — CHANGES_REQUESTED** [review]

**Comment by jeroen-hso** [comment]
@ahinkle my first integration test! I have applied your change request

**Comment by ahinkle** [comment]
Thanks

**Pull request merged**